frontend/apps/monastery/views/api400-designer/model/api400model.mjs

```diff
@@ -0,0 +1,209 @@
+/** 
+ * Model file for API400 application.
+ * (C) 2022 TekMonks. All rights reserved.
+ * License: See enclosed LICENSE file.
+ */
+import { algos } from "./algos.mjs";
+import { util } from "/framework/js/util.mjs";
+import { blackboard } from "/framework/js/blackboard.mjs";
+
+const EMPTY_MODEL = { apicl: [] }, DEFAULT_BUNDLE = "commands";
+let api400modelObj = EMPTY_MODEL, idCache = {}, current_command_bundle = DEFAULT_BUNDLE;
+const MSG_NODES_MODIFIED = "NODES_MODIFIED", MSG_CONNECTORS_MODIFIED = "CONNECTORS_MODIFIED",
+    MSG_NODE_DESCRIPTION_CHANGED = "NODE_DESCRIPTION_CHANGED", MSG_ARE_NODES_CONNECTABLE = "ARE_NODES_CONNECTABLE",
+    MSG_GET_MODEL = "GET_MODEL", MSG_RESET = "RESET", MSG_LOAD_MODEL = "LOAD_MODEL",
+    MSG_CONNECT_NODES = "CONNECT_NODES", MSG_ADD_NODE = "ADD_NODE" ;
+
+function init() {
+
+    blackboard.registerListener(MSG_NODES_MODIFIED, message => modelNodesModified(message.type, message.nodeName,
+        message.id, message.properties), true);
+    blackboard.registerListener(MSG_CONNECTORS_MODIFIED, message => modelConnectorsModified(message.type,
+        message.sourceNode, message.targetNode, message.sourceID, message.targetID));
+    blackboard.registerListener(MSG_NODE_DESCRIPTION_CHANGED, message => nodeDescriptionChanged(message.nodeName,
+        message.id, message.description));
+    blackboard.registerListener(MSG_ARE_NODES_CONNECTABLE, message => isConnectable(message.sourceName,
+        message.targetName, message.sourceID, message.targetID), true);
+    blackboard.registerListener(MSG_GET_MODEL, message => getModelAsFile(message.name), true);
+    blackboard.registerListener(MSG_RESET, _ => { api400modelObj = EMPTY_MODEL, idCache = {}, current_command_bundle = DEFAULT_BUNDLE; }, true);
+    blackboard.registerListener(MSG_LOAD_MODEL, message => loadModel(message.data));
+}
+
+function loadModel(jsonModel) {
```

Comment:
I still do NOT understand how this reads APICL. Seems the back parser is simply not written!

Reply:
We are calling the APICL parser funtion at the time of Upload it from local OR get it from server , in Open.mjs file.

We havn't this file for this release-1 PR , we will push it in release-2 today.

Reply:
OK